### PR TITLE
Guide to continuous deployment for a Django application

### PR DESCRIPTION
**Pull request by ervinb**

### CODE REVIEW

**Comment by markoa** [comment]
@ervinb Please don't use Firefox for screenshots, their select boxes look like it's 1995.

File: source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
+---
+layout: post
+title: Continuous Integration and Deployment of a Django application from GitHub to Heroku with Semaphore
+category: Python
+---
+
+In this guide, I will show you how to deploy a Django application to Heroku
```
Comment (markoa) [inline source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md]: s/I/we

File: source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+using Semaphore. I will be using a barebones, open-source [Django repository](https://github.com/ervinb/django_example)
+on GitHub. Feel free to fork it and use it to follow along if you don't have a
+project already set up.
+
+The sample application will be deployed automatically by Semaphore each time the
+master branch is built successfully. Semaphore let's us easily setup this kind of
```
Comment (markoa) [inline source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md]: every time all tests pass on the master branch

s/let's/lets, setup/set up

File: source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+The sample application will be deployed to Heroku. Both [Semaphore](https://semaphoreci.com/docs/python-continuous-integration.html)
+and [Heroku](https://devcenter.heroku.com/articles/getting-started-with-python#introduction)
+have native support for Python, so setting up a project using these services is
+a breeze.
+
+### Requirements
+
+  - Heroku account
+  - Semaphore account
```
Comment (markoa) [inline source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md]: Add "([create one here](https://semaphoreci.com/users/sign_up))"

File: source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  - Python with virtualenv installed on your machine
+  - git installed on your machine
+
+### Setting up continuous integration with Semaphore and GitHub
+
+After creating a Semaphore account and hooking it up with your GitHub or BitBucket
+account, you are ready to start. The next step is to [add](https://semaphoreci.com/docs/adding-github-bitbucket-project-to-semaphore.html)
+the Django project from the list of your available repositories.
```
Comment (markoa) [inline source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md]: This paragraph sounds like connecting Semaphore with GitHub is separate from the process of adding the project. You should simply merge these two sentences into a shorter one.

File: source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+<img src="/docs/assets/img/languages/python/deploy-django-to-heroku/django_project_list.png">
+
+In the following steps, selecting the branch and the account where to add the
+project will bring you to the analysis page, where Semaphore determines the
+settings for your project. These settings can be [changed at any time in
```
Comment (markoa) [inline source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md]: typo `[changed`

File: source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+[Project Settings](https://semaphoreci.com/docs/customizing-build-commands.html).
+
+<img src="/docs/assets/img/languages/python/deploy-django-to-heroku/django_analysis_complete.png">
+
+Completing the setup will trigger the first build.
```
Comment (markoa) [inline source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md]: s/trigger the/launch your

File: source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+<first-build-success>
+<img src="/docs/assets/img/languages/python/deploy-django-to-heroku/django_first_build.png">
+
+### Setting up continuous deployment from Semaphore to Heroku
+
+By installing [heroku-toolbelt](https://toolbelt.heroku.com/debian), you gain
+access to [Heroku CLI](https://github.com/heroku/heroku). With this tool, you can
+setup your application on Heroku without leaving the comfort of your terminal.
```
Comment (markoa) [inline source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md]: s/setup/set up

File: source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+will guide you through the process of setting up your first deployment to Heroku.
+
+<img src="/docs/assets/img/languages/python/deploy-django-to-heroku/django_deployment_destination.png">
+
+Select automatic deployment, as we want every successful build to trigger a new
+deploy.
+
+<img src="/docs/assets/img/languages/python/deploy-django-to-heroku/django_deployment_type.png">
+
+Select the branch which will be tied to the deployment.
```
Comment (markoa) [inline source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md]: Select the branch from which you would like to deploy.

File: source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+
+<img src="/docs/assets/img/languages/python/deploy-django-to-heroku/django_deployment_first.png">
+
+After the deployment completes, visiting the server's [URL](https://django-example-semaphore.herokuapp.com/)
+will present you the Django application running on Heroku.
+
+<img src="/docs/assets/img/languages/python/deploy-django-to-heroku/django_deployment_app_page.png">
+
+After completing these steps, your project has continuous integration and deployment
+set up and you can enjoy the [befefits](https://semaphoreci.com/community/tutorials/continuous-integration)
```
Comment (markoa) [inline source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md]: Better link "continuous integration and deployment" then "benefits" (semantics + SEO).

File: source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+---
+
+In this guide, I will show you how to deploy a Django application to Heroku
+using Semaphore. I will be using a barebones, open-source [Django repository](https://github.com/ervinb/django_example)
+on GitHub. Feel free to fork it and use it to follow along if you don't have a
+project already set up.
+
+The sample application will be deployed automatically by Semaphore each time the
+master branch is built successfully. Semaphore let's us easily setup this kind of
+deployment which is also known as continuous deployment. With this in place,
```
Comment (dunjardl) [inline source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md]: , which (missing comma)

File: source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+we can be sure that we have a working code on our server at all times.
```
Comment (dunjardl) [inline source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md]: have working code (remove "a", code is a noncount noun)

File: source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+and [Heroku](https://devcenter.heroku.com/articles/getting-started-with-python#introduction)
+have native support for Python, so setting up a project using these services is
+a breeze.
+
+### Requirements
+
+  - Heroku account
+  - Semaphore account
+  - Python with virtualenv installed on your machine
+  - git installed on your machine
```
Comment (dunjardl) [inline source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md]: Git (capitalisation)

File: source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+
+### Setting up continuous integration with Semaphore and GitHub
+
+After creating a Semaphore account and hooking it up with your GitHub or BitBucket
+account, you are ready to start. The next step is to [add](https://semaphoreci.com/docs/adding-github-bitbucket-project-to-semaphore.html)
+the Django project from the list of your available repositories.
+
+<img src="/docs/assets/img/languages/python/deploy-django-to-heroku/django_project_list.png">
+
+In the following steps, selecting the branch and the account where to add the
```
Comment (dunjardl) [inline source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md]: account to which you would like to add the

File: source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+deploy.
+
+<img src="/docs/assets/img/languages/python/deploy-django-to-heroku/django_deployment_type.png">
+
+Select the branch which will be tied to the deployment.
+
+<img src="/docs/assets/img/languages/python/deploy-django-to-heroku/django_deployment_branch.png">
+
+When asked for your Heroku API key, it can be found by going to [account settings](https://dashboard.heroku.com/account)
+on Heroku and clicking the "Show API Key" button on the bottom.
```
Comment (dunjardl) [inline source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md]: at the bottom.

File: source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+
+Comming back to Semaphore, select the Heroku application to which Semaphore will
```
Comment (dunjardl) [inline source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md]: Go back to Semaphore and select

File: source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+  - Python with virtualenv installed on your machine
+  - git installed on your machine
+
+### Setting up continuous integration with Semaphore and GitHub
+
+After creating a Semaphore account and hooking it up with your GitHub or BitBucket
+account, you are ready to start. The next step is to [add](https://semaphoreci.com/docs/adding-github-bitbucket-project-to-semaphore.html)
+the Django project from the list of your available repositories.
+
+<img src="/docs/assets/img/languages/python/deploy-django-to-heroku/django_project_list.png">
```
Comment (dunjardl) [inline source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md]: Please add borders and alt tags to all images - e.g. Django project list

File: source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md [hunk 17]
```diff
@@ -0,0 +1,108 @@
...
+---
+
+In this guide, we will show you how to deploy a Django application to Heroku
+using Semaphore. We will be using a barebones, open-source [Django repository](https://github.com/ervinb/django_example)
+on GitHub. Feel free to fork it and use it to follow along if you don't have a
+project already set up.
+
+The sample application will be deployed automatically by Semaphore every time all
+tests pass on the master branch. Semaphore lets us easily set up this kind of
+deployment which, is also known as continuous deployment. With this in place,
```
Comment (dunjardl) [inline source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md]: , which is (comma before which)

File: source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md [hunk 18]
```diff
@@ -0,0 +1,108 @@
...
+<img src="/docs/assets/img/languages/python/deploy-django-to-heroku/django_project_list.png" class="img-responsive" alt="Project list">
+
+In the following steps, selecting the branch and the account to which you wish
+to add the project will bring you to the analysis page, where Semaphore determines
+the settings for your project. These settings can be changed at any time in
+[Project Settings](https://semaphoreci.com/docs/customizing-build-commands.html).
+
+<img src="/docs/assets/img/languages/python/deploy-django-to-heroku/django_analysis_complete.png" class="img-responsive" alt="Semaphore analysis complete">
+
+Completing the setup will launch your the first build.
```
Comment (dunjardl) [inline source/docs/continuous-integration-deployment-django-with-github-semaphore-heroku.md]: your first build (remove the)

**Pull request merged**